src/proxy/http/canonicalize.rs

```diff
@@ -9,10 +9,12 @@
 //! DNS TTLs are honored and, if the resolution changes, the inner stack is
```

Comment:
This isn't true, is it?  I don't see the inner stack getting rebuilt.  As far as I can tell, this module just sets the canonicalized addr into the request extensions.